aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1494,11 +1488,92 @@ impl AptosVM {
         )
     }
 
+    fn charge_package_dependencies<'a>(
+        &self,
+        module_storage: &impl AptosModuleStorage,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext<'a>,
+        modules: &'a [CompiledModule],
+    ) -> Result<(), VMStatus> {
+        //
+        // Must exclude the ones that are in the current bundle because they have not
+        // been published yet.
+        let module_ids_in_bundle = modules
+            .iter()
+            .map(|module| (module.self_addr(), module.self_name()))
+            .collect::<BTreeSet<_>>();
+
+        // Not lazy loading: traverse all transitive dependencies and charge gas.
```

Comment:
let's mention recursion, because there is traversal in the below code but it just does one level.

Reply:
exactly